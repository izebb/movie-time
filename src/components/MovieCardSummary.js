import React from 'react';
import styled from 'styled-components';

const Title = styled.h4`
  margin-bottom: 4px;
  font-weight: 700;
  font-size: 17px;
  text-shadow: 0 1px 1px #000;
`;

const Sumamary = styled.div`
  transition: transform 120ms 50ms ease-out;
  transform: translate3d(0, 5px, 0);
`;

const Season = styled.div`
  margin-bottom: 2px;
`;

const Container = styled.div`
  position: absolute;
  left: 0;
  bottom: 0;
  width: 100%;
  color: #ffffff;
  height: 100%;
  padding: 20px 10px 30px 10px;
  background: linear-gradient(
    to bottom,
    rgba(30, 40, 45, 0.23) 0%,
    rgba(30, 40, 45, 0.35) 2%,
    rgba(30, 40, 45, 0.65) 50%,
    rgba(30, 40, 45, 0.65) 100%
  );
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  opacity: 0;
  transition: opacity 150ms ease-out;
  font-size: 15px;
  font-weight: 600;

  &:hover {
    opacity: 1;
    ${Sumamary} {
      transform: translate3d(0, 0, 0);
    }
  }
`;

export const MovieCardSummary = ({ title, seasons }) => {
  return (
    <Container>
      <Sumamary>
        <Title>{title}</Title>
        <Season>{seasons} Seasons</Season>
      </Sumamary>
    </Container>
  );
};
