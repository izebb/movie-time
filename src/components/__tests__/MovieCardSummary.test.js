import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MovieCardSummary } from '../MovieCardSummary';

describe('MovieCardSummary', () => {
  it('should render without crashing', () => {
    const props = {
      title: 'a beautiful mind',
      seasons: 5
    };
    const wrapper = shallow(<MovieCardSummary {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
