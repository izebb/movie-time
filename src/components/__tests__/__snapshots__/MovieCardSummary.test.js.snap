// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardSummary should render without crashing 1`] = `
<MovieCardSummary__Container>
  <MovieCardSummary__Sumamary>
    <MovieCardSummary__Title>
      a beautiful mind
    </MovieCardSummary__Title>
    <MovieCardSummary__Season>
      5
       Seasons
    </MovieCardSummary__Season>
  </MovieCardSummary__Sumamary>
</MovieCardSummary__Container>
`;
